### PR TITLE
Add documentation around gRPC-JSON transcoder effects on route matchers

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description

WIP....

# Context

I came across this behavior of the transcoder while looking into the behavior described in https://github.com/solo-io/gloo/issues/5742 . A second order affect of the transcoder's effects on route matchers is to prevent our rate-limiter plugin from operating as expected.

### CODE REVIEW

File: projects/gloo/api/v1/options/grpc_json/grpc_json.proto [hunk 1]
```diff
@@ -70,8 +70,10 @@ message GrpcJsonTranscoder {
     PrintOptions print_options = 3;
 
     // Whether to keep the incoming request route after the outgoing headers have been transformed to
-    // the match the upstream gRPC service. Note: This means that routes for gRPC services that are
-    // not transcoded cannot be used in combination with *match_incoming_request_route*.
+    // the match the upstream gRPC service.
+    // Note: When set to false, assumptions about the way that route matching is handled may be broken at the gateway/cluster level.
+    // In short, Envoy will not match in the incoming request path when this is false. See <https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/grpc_json_transcoder_filter#route-configs-for-transcoded-requests> for more details.
+    // Note: This means that routes for gRPC services that arr not transcoded cannot be used in combination with *match_incoming_request_route*.
```
[the remark this answers is not in the document]
Comment (ben-taussig-solo) [reply projects/gloo/api/v1/options/grpc_json/grpc_json.proto]: Corrected this in https://github.com/solo-io/gloo/pull/5833/commits/753a4539dc848898ca02baf033e7bc25f5069343

File: projects/gloo/api/v1/options/grpc_json/grpc_json.proto [hunk 2]
```diff
@@ -70,8 +70,10 @@ message GrpcJsonTranscoder {
     PrintOptions print_options = 3;
 
     // Whether to keep the incoming request route after the outgoing headers have been transformed to
-    // the match the upstream gRPC service. Note: This means that routes for gRPC services that are
-    // not transcoded cannot be used in combination with *match_incoming_request_route*.
+    // the match the upstream gRPC service.
+    // Note: When set to false, assumptions about the way that route matching is handled may be broken at the gateway/cluster level.
+    // In short, Envoy will not match in the incoming request path when this is false. See <https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/grpc_json_transcoder_filter#route-configs-for-transcoded-requests> for more details.
+    // Note: This means that routes for gRPC services that are not transcoded cannot be used in combination with *match_incoming_request_route*.
```
Comment (artberger) [inline projects/gloo/api/v1/options/grpc_json/grpc_json.proto:76]: Could we say which values trigger the behavior? I was a little uncertain what the last note meant. Maybe something like:

    // Set this value to true to keep the incoming request route after the outgoing headers are transformed to match the upstream gRPC service.
    // Note that you cannot set this value to true with routes for gRPC services that are not transcoded.
    // When set to false, Envoy does not match for the incoming request path.
    // For more information, see the Envoy docs <https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/grpc_json_transcoder_filter#route-configs-for-transcoded-requests>.

Comment (ben-taussig-solo) [reply projects/gloo/api/v1/options/grpc_json/grpc_json.proto:76]: Just applied this suggestion :)

**Review by artberger — APPROVED** [review]

**Pull request merged**